pkg/pullsync/pullsync.go

```diff
@@ -398,5 +410,17 @@ func (s *Syncer) cancelHandler(ctx context.Context, p p2p.Peer, stream p2p.Strea
 }
 
 func (s *Syncer) Close() error {
+	s.logger.Info("pull syncer shutting down")
+	close(s.quit)
+	cc := make(chan struct{})
+	go func() {
+		defer close(cc)
+		s.wg.Wait()
```

Comment:
Maybe this needs to wait for for loop in manage to exit before calling `wg.Wait()` as more go routines might be added?